src/statsUtils.js

```diff
@@ -0,0 +1,84 @@
+const {createWriteStream} = require('fs');
+const {Readable} = require('stream');
+
+const _ = require('lodash');
+
+class StatsSerializeStream extends Readable {
+  constructor(stats) {
+    super();
+    this._indentLevel = 0;
+    this._stringifier = this._stringify(stats);
+  }
+
+  get _indent() {
+    return '  '.repeat(this._indentLevel);
+  }
+
+  _read() {
+    let readMore = true;
+
+    while (readMore) {
+      const {value, done} = this._stringifier.next();
+
+      if (done) {
+        this.push(null);
+        readMore = false;
+      } else {
+        readMore = this.push(value);
+      }
+    }
+  }
+
+  * _stringify(obj) {
+    if (typeof obj === 'string' || typeof obj === 'number' || typeof obj === 'boolean' || obj === null) {
+      yield JSON.stringify(obj);
+    } else if (_.isArray(obj)) {
+      yield '[';
+      this._indentLevel++;
+
+      let isFirst = true;
+      for (let item of obj) {
+        if (item === undefined) {
+          item = null;
+        }
+
+        yield `${isFirst ? '' : ','}\n${this._indent}`;
+        yield* this._stringify(item);
+        isFirst = false;
+      }
+
+      this._indentLevel--;
+      yield obj.length ? `\n${this._indent}]` : ']';
+    } else if (_.isPlainObject(obj)) {
```

Comment:
There is the package which solves the problem problem
https://github.com/jonschlinkert/is-plain-object

Reply:
We use `lodash` in a lot of other places so there is no need to use other dependencies instead (at least for now).

Reply:
Actually, decided to remove this condition completely as it's not needed.